yb-voyager/src/queryparser/query_parser.go

```diff
@@ -163,4 +213,79 @@ func getCreateViewNode(parseTree *pg_query.ParseResult) (*pg_query.Node_ViewStmt
 func getCreateFuncStmtNode(parseTree *pg_query.ParseResult) (*pg_query.Node_CreateFunctionStmt, bool) {
 	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateFunctionStmt)
 	return node, ok
-}
\ No newline at end of file
+}
+
+func getCreateTableStmtNode(parseTree *pg_query.ParseResult) (*pg_query.Node_CreateStmt, bool) {
+	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateStmt)
+	return node, ok
+}
+
+func getCreateIndexStmtNode(parseTree *pg_query.ParseResult) (*pg_query.Node_IndexStmt, bool) {
+	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_IndexStmt)
+	return node, ok
+}
+
+func getAlterStmtNode(parseTree *pg_query.ParseResult) (*pg_query.Node_AlterTableStmt, bool) {
+	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_AlterTableStmt)
+	return node, ok
+}
+
+func getCreateTriggerStmtNode(parseTree *pg_query.ParseResult) (*pg_query.Node_CreateTrigStmt, bool) {
+	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateTrigStmt)
+	return node, ok
+}
+
+func getPolicyStmtNode(parseTree *pg_query.ParseResult) (*pg_query.Node_CreatePolicyStmt, bool) {
+	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreatePolicyStmt)
+	return node, ok
+}
+
+func getCompositeTypeStmtNode(parseTree *pg_query.ParseResult) (*pg_query.Node_CompositeTypeStmt, bool) {
+	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CompositeTypeStmt)
+	return node, ok
+}
+
+func getEnumTypeStmtNode(parseTree *pg_query.ParseResult) (*pg_query.Node_CreateEnumStmt, bool) {
+	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateEnumStmt)
+	return node, ok
+}
+func getForeignTableStmtNode(parseTree *pg_query.ParseResult) (*pg_query.Node_CreateForeignTableStmt, bool) {
+	node, ok := parseTree.Stmts[0].Stmt.Node.(*pg_query.Node_CreateForeignTableStmt)
+	return node, ok
+}
+
+func IsAlterTable(parseTree *pg_query.ParseResult) bool {
+	_, isAlter := getAlterStmtNode(parseTree)
+	return isAlter
+}
+
+func IsCreateIndex(parseTree *pg_query.ParseResult) bool {
+	_, isCreateIndex := getCreateIndexStmtNode(parseTree)
+	return isCreateIndex
+}
+
+func IsCreateTable(parseTree *pg_query.ParseResult) bool {
+	_, IsCreateTable := getCreateTableStmtNode(parseTree)
+	return IsCreateTable
+}
+
+func IsCreatePolicy(parseTree *pg_query.ParseResult) bool {
+	_, isCreatePolicy := getPolicyStmtNode(parseTree)
+	return isCreatePolicy
+}
+
+func IsCreateTrigger(parseTree *pg_query.ParseResult) bool {
+	_, isCreateTrig := getCreateTriggerStmtNode(parseTree)
+	return isCreateTrig
+}
+
+func IsCreateType(parseTree *pg_query.ParseResult) bool {
+	_, isComposite := getCompositeTypeStmtNode(parseTree)
+	_, isEnum := getEnumTypeStmtNode(parseTree)
+	return isComposite || isEnum
+}
+
+func IsCreateForeign(parseTree *pg_query.ParseResult) bool {
+	_, isForeignTable := getForeignTableStmtNode(parseTree)
+	return isForeignTable
+}
```

Comment:
Here we were dealing with protobuf messages instead of structs. 
The above code can be generalised and simplified like this(can be ::
```
// IsNodeType checks if the provided message is of the specified type.
func IsNodeType(msg protoreflect.Message, targetType string) bool {
	if msg == nil || !msg.IsValid() {
		return false
	}
	fmt.Printf("msgType: %s\n", GetMsgFullName(msg))
	return GetMsgFullName(msg) == targetType
}

// Predicate function to check for specific statement types
func IsStatementType(msg protoreflect.Message, nodeType string) bool {
	nodeMsg := getOneOfActiveField(msg, "node")
	if nodeMsg == nil {
		return false
	}
	return IsNodeType(nodeMsg, nodeType)
}

func IsCreateTable(msg protoreflect.Message) bool {
	return IsStatementType(msg, NodeCreateStmt)
}

func IsCreateView(msg protoreflect.Message) bool {
	return IsStatementType(msg, NodeViewStmt)
}

func IsCreateFunction(msg protoreflect.Message) bool {
	return IsStatementType(msg, NodeCreateFunctionStmt)
}

func IsCreateIndex(msg protoreflect.Message) bool {
	return IsStatementType(msg, NodeIndexStmt)
}

func IsAlterTable(msg protoreflect.Message) bool {
	return IsStatementType(msg, NodeAlterTableStmt)
}

func IsCreateTrigger(msg protoreflect.Message) bool {
	return IsStatementType(msg, NodeCreateTrigStmt)
}

func IsCreatePolicy(msg protoreflect.Message) bool {
	return IsStatementType(msg, NodeCreatePolicyStmt)
}

func IsCreateCompositeType(msg protoreflect.Message) bool {
	return IsStatementType(msg, NodeCompositeTypeStmt)
}

func IsCreateEnumType(msg protoreflect.Message) bool {
	return IsStatementType(msg, NodeCreateEnumStmt)
}

func IsCreateForeignTable(msg protoreflect.Message) bool {
	return IsStatementType(msg, NodeCreateForeignTableStmt)
}
```

Reply:
All the above added helper function can be further reduced to just have one helper function:
```
// Predicate function to check for specific statement types
func GetStatementType(msg protoreflect.Message) bool {
	nodeMsg := getOneOfActiveField(msg, "node")
	if nodeMsg == nil {
		return false
	}
	return GetMsgFullName(nodeMsg)
}
```

The main caller function can directly use that::
```
func GetDDLParser(parseTree *pg_query.ParseResult) (DDLParser, error) {
	switch GetStatementType(parseTree..Stmts[0].Stmt.ProtoReflect()) {
	case NODE_CreateTable:
		return NewTableParser(), nil
	case NODE_CreateIndex:
		return NewIndexParser(), nil
	case NODE_AlterTable:
		return NewAlterTableParser(), nil
	case NODE_CreatePolicy:
		return NewPolicyParser(), nil
	case NODE_CreateTrigger:
		return NewTriggerParser(), nil
	case NODE_CreateType:
		return NewTypeParser(), nil
	case NODE_CreateForeign:
		return NewForeignTableParser(), nil
	default:
		return NewNoOpParser(), nil
	}
}
```